### PR TITLE
[GLUTEN-5880][CORE] Ignore fallback for ColumnarWriteFilesExec children

### PR DESCRIPTION
**Pull request by wForget**
## What changes were proposed in this pull request?

As title

Fixes: #5880

## How was this patch tested?

added unit test and manual test:

```
create table wz_test_20240904 (a int) STORED AS PARQUET;
insert into wz_test_20240904 values (1);
```

before:
img_v3_02ed_6783be03-2d39-449d-bdf5-a8efceb2b39g

after this:
img_v3_02ed_8daaf8be-e484-43a8-9f65-a9f22c0cb34g

### CODE REVIEW

**Comment by wForget** [comment]
@Yohahaha @zhztheplayer could you please take a look?

File: gluten-ut/spark35/src/test/scala/org/apache/spark/sql/sources/GlutenInsertSuite.scala [hunk 1]
```diff
@@ -113,6 +114,13 @@ class GlutenInsertSuite
         spark.sparkContext.removeSparkListener(taskListener)
         spark.listenerManager.unregister(queryListener)
       }
+
+      // check no fallback nodes
+      val df2 = spark.sql("INSERT INTO TABLE pt2 SELECT * FROM pt")
+      checkWriteFilesAndGetChild(df2)
+      val fallbackSummary = GlutenImplicits
+        .collectQueryExecutionFallbackSummary(spark, df2.queryExecution)
+      assert(fallbackSummary.numFallbackNodes == 0)
```
Comment (Yohahaha) [inline gluten-ut/spark35/src/test/scala/org/apache/spark/sql/sources/GlutenInsertSuite.scala:123]: would you port the test to Spark34 module?

**Review by Yohahaha — APPROVED** [review]
LGTM! thank you

File: gluten-ut/spark35/src/test/scala/org/apache/spark/sql/sources/GlutenInsertSuite.scala (hunk 1 above)
Comment (wForget) [reply gluten-ut/spark35/src/test/scala/org/apache/spark/sql/sources/GlutenInsertSuite.scala:123]: sure, ported

**Review by zhztheplayer — APPROVED** [review]
👍

**Pull request merged**